metadata/src/utils/validation.rs

```diff
@@ -115,18 +115,14 @@ fn get_variant_hash(
 fn get_type_def_variant_hash(
     registry: &PortableRegistry,
     variant: &TypeDefVariant<PortableForm>,
-    only_these_variants: Option<&[&str]>,
     cache: &mut HashMap<u32, CachedHash>,
     outer_enum_hashes: &OuterEnumHashes,
 ) -> Hash {
     let variant_id_bytes = [TypeBeingHashed::Variant as u8; HASH_LEN];
     let variant_field_bytes = variant.variants.iter().fold([0u8; HASH_LEN], |bytes, var| {
         // With EncodeAsType and DecodeAsType we no longer care which order the variants are in,
         // as long as all of the names+types are there. XOR to not care about ordering.
-        let should_hash = only_these_variants
-            .as_ref()
-            .map(|only_these_variants| only_these_variants.contains(&var.name.as_str()))
-            .unwrap_or(true);
+        let should_hash = true;
```

Comment:
Nit: this owuld need tidying before merging :)

Reply:
done